File: Vision/Vision/FunctionModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Vision
{
    public interface FunctionModule
    {
        // Starts running the module and returns whether the module launches successfully.
        Boolean start();
        // Stops running the module.
        void stop();
        Boolean isHealthy();
        // Returns latest error message of this module, if any.
        String errorMessage();
    }
}

File: Vision/BankRobot/src/com/nb/robot/common/FunctionModule.java
package com.nb.robot.common;

public interface FunctionModule {	
	// Starts running the module and returns whether the module launches successfully.
	public boolean start();
	
	// Stops running the module.
	public void stop();

	// Returns if the module is healthy, like hardware is functional,
	// configuration file is valid.
	boolean isHealthy();

	// Returns latest error message of this module, if any.
	public String errorMessage();
}
